class-jetpack-wizard-banner.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Displays the first page of the Wizard in a banner form
+ *
+ * @package none
+ */
+
+use Automattic\Jetpack\Assets;
+use Automattic\Jetpack\Assets\Logo as Jetpack_Logo;
+
+/**
+ * Jetpack_Wizard_Banner
+ **/
+class Jetpack_Wizard_Banner {
+	/**
+	 * Jetpack_Wizard_Banner
+	 *
+	 * @var Jetpack_Wizard_Banner
+	 **/
+	private static $instance = null;
+
+	/**
+	 * Factory method
+	 */
+	public static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_Wizard_Banner();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Jetpack_Wizard_Banner constructor.
+	 */
+	private function __construct() {
+		add_action( 'current_screen', array( $this, 'maybe_initialize_hooks' ) );
+	}
+
+	/**
+	 * Initialize hooks to display the banner
+	 */
+	public function maybe_initialize_hooks() {
+		// We already display the wizard at the Jetpack area.
+		if ( strpos( get_current_screen()->id, 'jetpack' ) !== false ) {
+			return;
+		}
+
+		// Kill if banner has been dismissed.
+		if ( Jetpack_Options::get_option( 'dismissed_wizard_banner' ) ) {
+			return;
+		}
+
+		add_action( 'admin_print_styles', array( $this, 'admin_banner_styles' ) );
+		add_action( 'admin_notices', array( $this, 'render_banner' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_banner_scripts' ) );
+	}
+
+	/**
+	 * Enqueue JavaScript files.
+	 */
+	public static function enqueue_banner_scripts() {
+		wp_enqueue_script(
+			'jetpack-wizard-banner-js',
+			Assets::get_file_url_for_environment(
+				'_inc/build/jetpack-wizard-banner.min.js',
+				'_inc/jetpack-wizard-banner.js'
+			),
+			array( 'jquery' ),
+			JETPACK__VERSION,
+			true
+		);
+
+		wp_localize_script(
+			'jetpack-wizard-banner-js',
+			'jp_banner',
+			array(
+				'ajax_url'          => admin_url( 'admin-ajax.php' ),
+				'wizardBannerNonce' => wp_create_nonce( 'jp-wizard-banner-nonce' ),
+			)
+		);
+	}
+
+	/**
+	 * Include the needed styles
+	 */
+	public function admin_banner_styles() {
+		$min = ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? '' : '.min';
+
+		wp_enqueue_style(
+			'jetpack',
+			plugins_url( "css/jetpack-wizard-banner{$min}.css", JETPACK__PLUGIN_FILE ),
+			array(),
+			JETPACK__VERSION
+		);
+	}
+
+	/**
+	 * AJAX callback
+	 */
+	public static function ajax_callback() {
+		check_ajax_referer( 'jp-wizard-banner-nonce', 'nonce' );
+
+		if ( isset( $_REQUEST['dismissBanner'] ) ) {
+			Jetpack_Options::update_option( 'dismissed_wizard_banner', 1 );
+			wp_send_json_success();
+		}
+
+		wp_die();
+	}
+
+	/**
+	 * Renders the Wizard Banner
+	 */
+	public function render_banner() {
+		$jetpack_logo = new Jetpack_Logo();
+
+		?>
+		<div id="jp-wizard-banner" class="jp-wizard-banner">
```

Comment:
Given that this is replicating the contents of `_inc/client/setup-wizard/intro-page/index.jsx`, I think we should have a warning for now, in both files, that if you update one you must update the other as well.

On the long run, I think it would be nice if that content were retrieved from a single source to avoid duplicating things.